src/main/java/com/a2/backend/service/impl/TagServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package com.a2.backend.service.impl;
+
+import com.a2.backend.entity.Tag;
+import com.a2.backend.repository.TagRepository;
+import com.a2.backend.service.TagService;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.stereotype.Service;
+
+@Service
+public class TagServiceImpl implements TagService {
+
+    private final TagRepository tagRepository;
+
+    public TagServiceImpl(TagRepository tagRepository) {
+        this.tagRepository = tagRepository;
+    }
+
+    @Override
+    public List<Tag> createTagsList(List<String> tagsToAdd) {
```

Comment:
Optional: you could rename this to something like "findOrCreateTag" to specify what the method does